### PR TITLE
wrap MCintegration.jl

### LINKED ISSUE #194: Warp MCIntegration.jl
https://github.com/numericalEFT/MCIntegration.jl

### PR DESCRIPTION
**Pull request by lxvm**
Fixes #194 

This pr adds a VEGASMC algorithm from MCIntegration.jl

This library supports:
- vector-valued integrands
- inplace integrands
This library doesn't support:
- absolute or relative tolerances
- batched integrands

I think some of the tests may not pass due to underconvergence, which happens randomly due to MC sampling.

### CODE REVIEW

**Comment by frankier** [comment]
I guess you can seed your rng to control convergence, but QMC might also be helpful https://github.com/numericalEFT/MCIntegration.jl/issues/53

**Comment by ChrisRackauckas** [comment]
Make the tolerances a bit lower? If that's not helping then its stopping heuristic may be incorrect.

**Comment by kunyuan** [comment]
I'm the developer of MCIntegration.jl. Thank you for considering our package. If you need any help to fix the interface errors, please let me know :-)

**Comment by ChrisRackauckas** [comment]
@kunyuan I think the only major thing coming up is that it would be nice for it to directly use QuasiMonteCarlo as suggested here https://github.com/SciML/Integrals.jl/issues/207

File: Project.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ version = "4.1.0"
 CommonSolve = "38540f10-b2f7-11e9-35d8-d573e4eb0ff2"
 HCubature = "19dc6840-f33b-545b-b366-655c7e3ffd49"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MCIntegration = "ea1e2de9-7db7-4b42-91ee-0cd1bf6df167"
 MonteCarloIntegration = "4886b29c-78c9-11e9-0a6e-41e1f4161f7b"
```
[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply Project.toml:11]: 👍 sounds good.

**Comment by ChrisRackauckas** [comment]
> @ChrisRackauckas I saw https://github.com/numericalEFT/MCIntegration.jl/issues/51 so I moved MCIntegration functionality to an extension. It will be easy and non-breaking to migrate it to a dependency as soon as it is ready.

That's good thanks. Yeah, until MPI is an extension in MCIntegration.jl it probably cannot be brought in as a standard dependency since MPI.jl can cause build problems on some platforms.

**Review by ChrisRackauckas — APPROVED** [review]

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**